diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/LongNumericalValuesSeparatedRule.kt

```diff
@@ -0,0 +1,153 @@
+package org.cqfn.diktat.ruleset.rules
+
+import com.pinterest.ktlint.core.KtLint
+import com.pinterest.ktlint.core.Rule
+import com.pinterest.ktlint.core.ast.ElementType.FLOAT_LITERAL
+import com.pinterest.ktlint.core.ast.ElementType.INTEGER_LITERAL
+import org.cqfn.diktat.common.config.rules.RuleConfiguration
+import org.cqfn.diktat.common.config.rules.RulesConfig
+import org.cqfn.diktat.common.config.rules.getRuleConfig
+import org.cqfn.diktat.ruleset.constants.Warnings
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.impl.source.tree.LeafPsiElement
+import java.lang.StringBuilder
+
+class LongNumericalValuesSeparatedRule : Rule("long-numerical-values") {
+    private lateinit var configRules: List<RulesConfig>
+    private lateinit var emitWarn: ((offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit)
+    private var isFixMode: Boolean = false
+
+    companion object {
+        private const val DELIMITER_LENGTH: Int = 3
+        private const val MAX_NUMBER_LENGTH: Int = 3
+    }
+
+    override fun visit(node: ASTNode,
+                       autoCorrect: Boolean,
+                       params: KtLint.Params,
+                       emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit) {
+        configRules = params.getDiktatConfigRules()
+        emitWarn = emit
+        isFixMode = autoCorrect
+
+        val configuration = LongNumericalValuesConfiguration(
+                configRules.getRuleConfig(Warnings.LONG_NUMERICAL_VALUES_SEPARATED)?.configuration ?: mapOf())
+
+        if (node.elementType == INTEGER_LITERAL) {
+            if(!isValidIntegerConstant(node.text, configuration, node)) {
+                Warnings.LONG_NUMERICAL_VALUES_SEPARATED.warnAndFix(configRules, emitWarn, isFixMode, node.text, node.startOffset) {
+                    fixIntegerConstant(node, configuration.maxBlockLength)
+                }
+            }
+        }
+
+        if (node.elementType == FLOAT_LITERAL) {
+            if (node.text.contains("_")) {
```

Comment:
Please either move this check inside separate method, like for integers, or extract it from `isValidIntegerConstant`, so that two cases look similar. Actually, you can use same method for integer and both parts of float.